xunit-autolabeler-v2/ast_parser/core/polyglot_parser.py

```diff
@@ -45,15 +46,18 @@ def _recursor(method: pdd.PolyglotDriftData) -> pdd.PolyglotDriftData:
         Returns:
             An updated PolyglotDriftData object
         """
-        for child in method.children:
-            child_methods = [child_method for child_method in source_methods
-                             if 'name' in child_method]
-            child_methods = [child_method for child_method in child_methods
-                             if child == child_method['name']]
 
-            # prevent infinite loops
-            child_methods = [child_method for child_method in child_methods
-                             if child_method['name'] != method['name']]
+        # Prevent infinite loops
+        method_key = (method.name, method.class_name, method.source_path)
+        if method_key in recursed_nodes:
+            return
+        recursed_nodes.add(method_key)
+
+        for child in method.children:
+            child_methods = [
+                child_method for child_method in source_methods if
+                child_method.name and child == child_method.name
```

Comment:
Do you need to check child_method.name truthiness if you are then checking if it's equal to child?

Reply:
I was wondering the same thing. Apparently not, since tests pass _without_ this check.